src/charm.py

```diff
@@ -190,18 +201,17 @@ def ensure_state(self):
         )
 
     def handle_ingress(self):
-        ingress = self._get_interface("ingress")
-
-        if ingress:
-            for app_name, version in ingress.versions.items():
-                data = {
-                    "prefix": "/dex",
-                    "rewrite": "/dex",
-                    "service": self.model.app.name,
-                    "port": self.model.config["port"],
-                }
+        interfaces = self._get_interfaces()
+
+        if interfaces["ingress"]:
+            data = {
+                "prefix": "/dex",
+                "rewrite": "/dex",
+                "service": self.model.app.name,
+                "port": self.model.config["port"],
+            }
```

Comment:
Was the previous `._get_interface()` breaking here?  I'm not sure why we're switching to getting all interfaces

The downside to getting all interfaces is that any incomplete relation will show up here as a problem with the ingress relation.  The previous way that accessed just a single interface at a time led to cleaner error messages.

Reply:
If we do switch to getting all interfaces, we should be consistent and do it with the other oidc relation too, then remove the `_get_interface` helper.  But I'm inclined to keep the old way unless there was a reason to change